src/components/Tables/Venues.tsx

```diff
@@ -1,66 +1,141 @@
-import { useMemo, useState } from 'react'
+import { useCallback, useMemo, useState } from 'react'
 import { api } from '~/utils/api'
 import {
-    ColumnDef,
     flexRender,
     getCoreRowModel,
     useReactTable,
     SortingState,
     getSortedRowModel,
     getFilteredRowModel,
-    ColumnFiltersState
+    ColumnFiltersState,
+    createColumnHelper
 } from '@tanstack/react-table'
 import { Venue } from '~/types/data'
-import { Table, Box, Flex } from '@radix-ui/themes'
-import { HeaderCell } from './components'
+import { Table, Box, Flex, Text } from '@radix-ui/themes'
+import { HeaderCell, TableActionMenu } from './components'
 import Loading from '../Loading'
 import { fuzzyFilter } from './utils/filters'
 import { PaginationButtonGroup } from './components/PaginationButtonGroup'
+import FadeOutText from '../FadeOutText'
+import { useRouter } from 'next/router'
+
+const columnHelper = createColumnHelper<Venue>()
 
 export function VenuesTable() {
-    const { data, isFetched, isLoading } = api.venue.getAll.useQuery()
+    const { data, isFetched, isLoading, refetch } = api.venue.getAll.useQuery()
+    const router = useRouter()
+    const { mutate: setFeaturedMutation, isSuccess: setFeaturedIsSuccess } =
+        api.venue.setFeatured.useMutation()
+    const { mutate: deleteMutation, isSuccess: deleteMutationIsSuccess } =
+        api.venue.delete.useMutation()
+    const [error, setError] = useState<string | null>(null)
+
+    const handleEditClick = useCallback(
+        async (venue: Venue) => {
+            const params = new URLSearchParams()
+            params.set('id', venue.id)
+            await router.push(
+                {
+                    pathname: '/admin/edit-venue',
+                    query: params.toString()
+                },
+                undefined,
+                { shallow: true }
+            )
+        },
+        [router]
+    )
+
+    const handleToggleFeatured = useCallback(
+        (venue: Venue) => {
+            setFeaturedMutation(
+                { id: venue.id },
+                {
+                    onSuccess: () => {
+                        setError(null)
+                        void refetch()
+                    },
+                    onError: (e) => {
+                        setError(
+                            'Toggle featured failed. There was an error altering the database.'
+                        )
+                        console.error(e)
+                    }
+                }
+            )
+        },
+        [setFeaturedMutation, refetch]
+    )
+
+    const handleDelete = useCallback(
+        (venue: Venue) => {
+            deleteMutation(
+                { id: venue.id },
+                {
+                    onSuccess: () => {
+                        setError(null)
+                        void refetch()
```

Comment:
why do we need `void` here?

Reply:
Can't remember why I put those there. Im sure they fixed something some time.

Reply:
Ok now I remember: these are promises and so they need the void keyword in this context. Ommiting caused the build to fail.

Reply:
We need to change our linting rules in this case. I can do this in a follow-up task.